bootstrap/kubeadm/internal/ignition/clc/clc.go

```diff
@@ -0,0 +1,146 @@
+package clc
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"text/template"
+
+	ignition "github.com/coreos/ignition/config/v2_3"
+	ignitionTypes "github.com/coreos/ignition/config/v2_3/types"
+	clct "github.com/flatcar-linux/container-linux-config-transpiler/config"
+	"github.com/pkg/errors"
+
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/bootstrap/kubeadm/internal/cloudinit"
+)
+
+const (
+	clcTemplate = `---
+systemd:
+  units:
+    - name: kubeadm.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description=kubeadm
+        [Service]
+        # To not restart the unit when it exits, as it is expected.
+        Type=oneshot
```

Comment:
Having this as a oneshot is fine - we should however think about the outcome of the unit failing. Do you currently have an idea how such a case should be handled? 

Additionally WDYT about setting `RemainAfterExit=true` IIRC it made debugging a bit easier for us, but I might be misremembering.

Reply:
I initially had `RemainAfterExit=true`, but then decided it's not really needed. One can add it using a drop-in in using additional config when needed.

Reply:
+1 on drop-in. We can even completely disable the "default" behavoir.

Reply:
@MarcelMue mind if I mark it as resolved then?